src/server/modes/charts/plugins/datalens/preparers/line-time/index.ts

```diff
@@ -224,6 +225,14 @@ function prepareLineTime(options: PrepareFunctionArgs) {
             graph.color = colorData[i];
             graph.spanGaps = yPlaceholderSettings.nulls === 'connect';
         });
+
+        if (colorData.length > 1 && shared.extraSettings?.legendMode !== 'hide') {
```

Comment:
The legend display should be based on the number of series with different names, not the presence of color